doc/develop/extending.rst

```diff
@@ -620,7 +712,25 @@ of the Odemis GUI:
    # After every interesting call
    tr.print_diff()
 
-As it will not detect C library memory allocations, if no new large object has
+As it will not detect ``C`` library memory allocations, if no new large object has
 appeared and the Python process uses more memory, then it's likely a C library
 memory leak.
 
+To test numpy arrays for memory usage, it is possible to call::
+
+    numpy.ndarray.nbytes
+
+It displays the total bytes consumed by the elements of the array.
+It does not include memory consumed by non-element attributes of the array object.
+
+A similar and more generic way is to use the sys function to check on the memory allocated to your
+object of interest::
+
+    sys.getsizeof(yourObject)
+
+It returns the size of an object in bytes. The object can be any type of object.
+All built-in objects will return correct results, but this does not have to hold true for
+third-party extensions as it is implementation specific.
```

Comment:
This "third-party extensions" is basically the C-related memory allocation. I'd suggest to remove that last paragraph line, and add here:
`In other words, for objects created via a C library, the reported size might be correct, or might be underestimated.`